### PR TITLE
Enable DataDog etcd integration

### PR DESCRIPTION
**Pull request by michaljurecko**
Part of: https://keboola.atlassian.net/browse/BA-15

----- 
Changes:
- Enabled integration of the etcd with the DataDog, so metrics are collected in the DataDog.

### CODE REVIEW

File: .github/workflows/templates-api.yml [hunk 1]
```diff
@@ -40,9 +41,49 @@ jobs:
       - name: Deploy the new version, from the HEAD branch
         run: ./deploy_local.sh
       - name: Check deployment of the etcd
+        if: always()
         run: kubectl rollout status sts/templates-api-etcd --namespace templates-api
       - name: Check deployment of the API
+        if: always()
         run: kubectl rollout status deployment/templates-api --namespace templates-api
+      - name: Check access to the etcd from a client
```
[the remark this answers is not in the document]
Comment (michaljurecko) [reply .github/workflows/templates-api.yml:46]: Po spusteni to dopadlo takto, kedze DataDog pridavam do network policy az v dalsom commite:
image

https://github.com/keboola/keboola-as-code/actions/runs/3375806224

File: provisioning/templates-api/kubernetes/templates/etcd-values.yaml [hunk 2]
```diff
@@ -47,6 +47,14 @@ service:
...
   enabled: true
   allowExternal: false
+  extraIngress:
+   # Allow DataDog HTTP calls to the /metrics
+   - ports:
+       - port: 2379
+     from:
+       - podSelector:
+           matchLabels:
+             app: datadog-agent
```
Comment (michaljurecko) [inline provisioning/templates-api/kubernetes/templates/etcd-values.yaml:57]: Povolil som pristup k etcd z DataDog Agent.
Label som nasiel tu: https://github.com/keboola/infrastructure/blob/a0aaca1343113f4aa8ee5e3ab534490f7a887239/kubernetes/datadog/aws-helm-datadog-agent.yaml#L633

File: .github/workflows/templates-api.yml [hunk 3]
```diff
@@ -18,9 +19,12 @@ jobs:
...
     runs-on: ubuntu-latest
     steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
       - name: Start MiniKube
-        id: minikube
-        uses: medyagh/setup-minikube@latest
+        run: ./start_minikube.sh
```
Comment (michaljurecko) [inline .github/workflows/templates-api.yml:27]: Pouzil som radsej script, ktory je sucastou aj `deploy_local.sh`, kedze sa tam nedalo nastavit vsetko co som potreboval, vid script.

File: provisioning/templates-api/docker/Dockerfile [hunk 4]
```diff
@@ -2,9 +2,16 @@
...
+
+COPY Makefile Makefile
+COPY scripts scripts
 RUN make tools
+
+COPY go.mod go.mod
+COPY go.sum go.sum
+RUN go mod download
+
+COPY . .
```
Comment (michaljurecko) [inline provisioning/templates-api/docker/Dockerfile:14]: Pre rychlejsie lokalne testovanie, tools a dependencies sa nemusia stahovat pri kazdej zmene v kode.

File: provisioning/templates-api/start_minikube.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+echo "--------------------------"
+
+# 1. Use v1.24 - https://github.com/kubernetes/minikube/issues/15173
+# 2. Enable Network Policies support.
+#    Kindnet - default CNI, does not support Network Policies, by design.
+#    https://minikube.sigs.k8s.io/docs/handbook/network_policy/
+minikube start \
+--wait "apiserver,system_pods,default_sa,apps_running,node_ready,kubelet" \
+--kubernetes-version=v1.24.7 \
+--cni=calico
```
Comment (michaljurecko) [inline provisioning/templates-api/start_minikube.sh:15]: - `--wait`, niektore casti `MiniKube` startuju dhlsie - nase app pods sa zbytocne restartovali (nebolo potom jasne ci je to chyba alebo nie)
- `--kubernetes-version` - `v1.25` ma nejaky bug s `Calico` https://github.com/kubernetes/minikube/issues/15173
- `--cni=calico` MiniKube by default ignoruje `NetworkPolicy`, `calico` CNI ich podporuje, ... teda som to zapol, aby sa dalo co najviac veci otestovat lokalne a v CI

File: provisioning/templates-api/kubernetes/templates/etcd-values.yaml (hunk 2 above)
Comment (michaljurecko) [reply provisioning/templates-api/kubernetes/templates/etcd-values.yaml:57]: A testy uz presli:
image

https://github.com/keboola/keboola-as-code/actions/runs/3375869489/jobs/5602942462

**Review by JakubMatejka — APPROVED** [review]
za mě 👍

**Comment by Halama** [comment]
Jdu mrknout.

**Review by Halama — APPROVED** [review]
Vypadá to dobře 👍

**Pull request merged**